packages/controls/src/inputs/MaskedInput/MaskedInput.js

```diff
@@ -110,19 +111,34 @@ export default class MaskedInput extends Component {
 	}
 
 	handleComplete = ev => {
-		if (this.props.onChange) this.props.onChange(ev.target.value);
+		if (this.props.onChange) this.handleChange(ev);
 	};
 
 	handleCleared = () => {
 		if (this.props.onChange) this.props.onChange('');
 	};
 
 	handleIncomplete = ev => {
-		if (this.props.onChange) this.props.onChange(ev.target.value);
+		if (this.props.onChange) this.handleChange(ev);
+	};
+
+	handleChange = ev => {
+		const {type, onChange} = this.props;
+
+		switch (true) {
+			case type === 'number':
+				onChange(parseInt(ev.target.value, 10));
+				break;
+			case typeof type === 'function':
+				onChange(type(ev.target.value));
+				break;
+			default:
+				onChange(ev.target.value);
+		}
```

Comment:
This isn't quite right. `type` should never be `"function"`.

Make a function like this (it doesn't need to be a class method, it can be outside the class):

```js
function getConvertFunction(type) {
  if (isFunction(type)) return type;
  switch (type) {
    case 'number': return (str) =>parseInt(ev.target.value, 10);
    case 'string':
    default:
      return (str) => str;
  };
}
```

Then you need to modify `handleComplete`, `handleCleared` and `handleIncomplete` to look something like this:

```js
if (this.props.onChange) this.props.onChange(getConvertFunction(this.props.type)(ev.target.value));
```
(except handleCleared should pass in an empty string, instead of `ev.target.value`)

(Also make sure that the `number` type works under the condition that input is empty or a blank string.)

Reply:
Lots of discussion on the whole `switch(true)` idiom.

https://stackoverflow.com/questions/2765981/javascript-switchtrue
https://stackoverflow.com/questions/20596572/switch-true-as-alternative-to-else-if
https://stackoverflow.com/questions/14118996/is-switchtrue-valid-javascript

While it's definitely argued for being a possible alternative to `if() else if () ...`, I would personally hesitate in using it. On first read It's not clear what's going on. Things like that are generally avoided if possible.